env-var-docs/README.md

```diff
@@ -0,0 +1,69 @@
+# Environment Variable Documentation
+
+The main configuration file for the CiviForm server is in
+[application.conf](../server/conf/application.conf). The file is in [HOCON
```

Comment:
I agree some documentation on the layers is worth adding.  Because the env-var-docs is the inner-most layer, I don't think it should have specific details about civiform_config.sh.  I added the following paragraph:

The documentation includes support for adding value validation rules. Tools
that deploy the CiviForm server
([civiform/cloud-deploy-infra](https://github.com/civiform/cloud-deploy-infra),
for example) should use these rules to validate user-provided values before the
server is deployed.